Base/home/anon/tests/run-tests-and-shutdown.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+echo "==== Running Tests on SerenityOS ===="
+
+run() {
+    test_cmd=($*)
+    echo "Running test $idx/$count -- $test_cmd"
+    if $test_cmd {
+        echo "::info file=$test_cmd:: $test_cmd passed!"
+    } else {
+        echo "::error file=$test_cmd:: $test_cmd returned non-zero exit code, check logs!"
+    }
+    idx=$(expr $idx + 1)
+}
+
+# TODO: It'd be nice to have a list+list op (as opposed to nest-on-in-another)
+system_tests=(test-js (test-crypto bigint -t)) #("test-js" "test-web" "test-pthread" "test-crypto" "test-compress" $usr_tests_tests)
+all_tests=($system_tests $(find /usr/Tests -type f | grep -v Kernel | grep -v .inc | shuf))
+
+count=0
+for list in $all_tests {
+    for $list { count=$(expr $count + 1) }
+}
```

Comment:
Where is this count being used?

Reply:
It's used to print "Running test $name ($idx/ $count )" in the run() function above. Gotta love globals that come after the function they're used in right? Anyway the entire reason this loop is here is b/c there's no easy way to get a list's length without .. whatever this is. spawning a subshell to increment a value. Takes forever without KVM on :)

Reply:
Sounds like there should be shell built-ins "enumerate" (like in python) and "length" (of a list), CC @alimpfard :P

Reply:
> Takes forever without KVM on :)

It doesn't fix the underlying problem, but it's a good idea anyway: Can we switch KVM on? I do realize it probably won't work (the GA image is, itself, already a VM, so running a VM inside a VM probably won't have hardware-acceleration available), but is there a way to try?

Reply:
Yes, I am to blame for this.
it's not that the shell can't tell you the length of a list, it is that I cannot find the right syntax to add to the shell language.
I hate the sh `${#var}` syntax, (though I've had ideas about parsing this stuff to the existing AST)

I've experimented with `${length $var}` before (there's an open draft PR with that), but I didn't quite like that either.

re `enumerate`, not a _bad_ idea, but again, syntax how?

Reply:
> Can we switch KVM on?

Sadly no, because Github actions does not support nested virtualization.

Reply:
All right, i've removed the count and left a todo in its place :)